cluster/azure/keyvault_flexvol/main.tf

```diff
@@ -2,22 +2,9 @@ module "common-provider" {
   source = "../../common/provider"
 }
 
-data "azuread_service_principal" "flexvol" {
-  application_id = "${var.service_principal_id}"
-}
-
-resource "azurerm_key_vault_access_policy" "flexvol" {
-  count = "${var.enable_flexvol ? 1 : 0}"
-
-  vault_name          = "${var.keyvault_name}"
+data "azurerm_key_vault" "flexvol" {
+  name                = "${var.keyvault_name}"
   resource_group_name = "${var.resource_group_name}"
-
-  tenant_id = "${var.tenant_id}"
-  object_id = "${data.azuread_service_principal.flexvol.id}"
-
-  key_permissions         = "${var.flexvol_keyvault_key_permissions}"
-  secret_permissions      = "${var.flexvol_keyvault_secret_permissions}"
-  certificate_permissions = "${var.flexvol_keyvault_certificate_permissions}"
 }
 
 resource "null_resource" "deploy_flexvol" {
```

Comment:
`deploy_flexvol.sh` doesn't need `kvcreds` secret anymore since Managed Identity will be used instead of Service Principal. Consequentially, all service principal input parameters can also be removed in call to script `deploy_flexvol.sh`.